projects/demo/src/modules/components/dialog/examples/9/pay-modal/pay-modal.component.ts

```diff
@@ -0,0 +1,161 @@
+    tuiCardNumberValidator,
+    tuiDefaultCardValidator,
+    TuiInputCardGroupedComponent,
+} from '@taiga-ui/addon-commerce';
+import {TuiDestroyService} from '@taiga-ui/cdk';
+import {TuiDialogContext} from '@taiga-ui/core';
+import {POLYMORPHEUS_CONTEXT} from '@tinkoff/ng-polymorpheus';
+import {BehaviorSubject} from 'rxjs';
+import {map, switchMap, takeUntil} from 'rxjs/operators';
+
+import {
+    AccountCard,
+    DataForPayCardModal,
+    FetchedCards,
+    PaymentMode,
+} from '../helpers/models';
+import {PayService} from '../helpers/pay.service';
+import {inputCardGroupedCVCValidator} from '../helpers/validator';
+
+@Component({
+    selector: 'pay-modal',
+    templateUrl: './pay-modal.component.html',
+    styleUrls: ['./pay-modal.component.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [TuiDestroyService],
+})
+export class PayModalComponent implements OnInit {
+    @ViewChild('cardGroupedInput')
+    private readonly cardGroupedInput?: TuiInputCardGroupedComponent;
+
+    private savedCards: AccountCard[] = [];
+
+    readonly form = new FormGroup({
+        card: new FormControl(null, [
+            Validators.required,
+            inputCardGroupedCVCValidator(),
+        ]),
+        saveCard: new FormControl(true),
+    });
+
+    paymentMode: PaymentMode = PaymentMode.ByNewCard;
+    loading$ = new BehaviorSubject(false);
+    readonly PAYMENT_MODE = PaymentMode;
+    amount: number = this.context?.data?.amount ?? 0;
+    savedCardsWithoutSelected: AccountCard[] = [];
+    payProcessing = false;
+    selectedCardId: string | null = null;
+
+    constructor(
+        @Inject(POLYMORPHEUS_CONTEXT)
+        readonly context: TuiDialogContext<void, DataForPayCardModal>,
+        @Inject(PayService) private readonly payService: PayService,
+        @Self() @Inject(TuiDestroyService) private readonly destroy$: TuiDestroyService,
+    ) {}
+
+    ngOnInit(): void {
+        this.fetchCardsAndSetPrimaryCard();
+    }
+
+    setCard(savedCard?: AccountCard | null): void {
+        if (savedCard) {
+            this.form.patchValue({
+                card: {
+                    card: this.getSavedCardMaskedNumber(savedCard),
+                    expire: '**/**',
+                    cvc: '',
+                },
+            });
+
+            this.form.controls.card.removeValidators(tuiCardNumberValidator);
+            this.selectedCardId = savedCard.cardId;
+            this.paymentMode = PaymentMode.BySavedCard;
+            this.savedCardsWithoutSelected = this.savedCards.filter(
+                (card: AccountCard) => card.cardId !== this.selectedCardId,
+            );
+
+            this.focusCVC();
+        } else {
+            this.form.patchValue({card: null});
+            this.form.controls.card.addValidators(tuiCardNumberValidator);
+            this.selectedCardId = null;
+            this.savedCardsWithoutSelected = this.savedCards;
+            this.paymentMode = PaymentMode.ByNewCard;
```

Comment:
**nitpicking:**
It is a cool example, but it is little bit overcomplicated for documentation.

Can we simplify it ?

Reply:
fix